src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -935,6 +961,75 @@ impl Step for LldWrapper {
     }
 }
 
+/// Builds the `wasm-component-ld` linker wrapper, which is shipped with rustc to be executed on the
+/// host platform where rustc runs.
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct WasmComponentLd {
+    build_compiler: Compiler,
+    target: TargetSelection,
+}
+
+impl WasmComponentLd {
+    /// Returns `WasmComponentLd` that should be **used** by the passed compiler.
+    pub fn for_compiler(builder: &Builder<'_>, target_compiler: Compiler) -> Self {
+        Self {
+            build_compiler: get_tool_target_compiler(
+                builder,
+                ToolTargetBuildMode::Dist(target_compiler),
+            ),
+            target: target_compiler.host,
+        }
+    }
+}
+
+impl Step for WasmComponentLd {
+    type Output = ToolBuildResult;
+
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/wasm-component-ld")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(WasmComponentLd {
+            build_compiler: get_tool_target_compiler(
+                run.builder,
+                ToolTargetBuildMode::Build(run.target),
+            ),
+            target: run.target,
+        });
+    }
+
+    #[cfg_attr(
+        feature = "tracing",
+        instrument(
+            level = "debug",
+            name = "WasmComponentLd::run",
+            skip_all,
+            fields(build_compiler = ?self.build_compiler),
+        ),
+    )]
+    fn run(self, builder: &Builder<'_>) -> ToolBuildResult {
+        builder.ensure(ToolBuild {
+            build_compiler: self.build_compiler,
+            target: self.target,
+            tool: "wasm-component-ld",
+            mode: Mode::ToolTarget,
+            path: "src/tools/wasm-component-ld",
+            source_type: SourceType::InTree,
+            extra_features: vec![],
+            allow_features: "min-specialization",
```

Comment:
Question: does `wasm-component-ld` actually need `min-specialization`? At least, I'm not seeing it used at all.

Reply:
It compiled without it. @alexcrichton Do you know if this feature is required for `wasm-component-ld`?

Reply:
Ah, the historical reason is at https://github.com/tkaitchuck/aHash/issues/238

Reply:
Thank you for the context! I removed it, let's see what CI says.

Reply:
Wait hold on, that means this is necessary because ahash will try to enable it because of fragile can-enable-feature detection (I think?)

Reply:
I was able to build it locally without the feature, IIRC. So maybe the lockfile has been updated to a newer ahash version in the meantime? I remember that ahash eventually gave up with these detections. We'll see what happens on CI.

Reply:
I still see min_spec feature detection in ahash master's build.rs, hm

Reply:
I did `cargo tree` on `wasm-component-ld` and it doesn't return `ahash` anymore, so maybe that's the reason =D